query/src/main/java/org/infinispan/search/mapper/work/impl/SearchIndexerImpl.java

```diff
@@ -43,10 +48,16 @@ public CompletableFuture<?> add(Object providedId, String routingKey, Object ent
          return CompletableFutures.completedNull();
       }
 
-      return delegate.add(convertedValue.typeIdentifier, providedId,
```

Comment:
@jabolina I wouldn't serialize and block the indexing operations. At the moment we can use the I/IO thread (non blocking) to add the indexing operations to the internal blocking queue of Hibernate Search, that is quite efficient (even if not optimal -- but Lucene indexing operations are still blocking). I wound avoid to Introduce a `synchronized` and switch the thread to blocking one.
If the goal here is to stop the test only if there is no ongoing indexing operations, I think that there is other way. We could for instance introduce an SPI to await on it. If you agree I can handle it, but later this week.

Reply:
I updated the `RemoteHitCountAccuracyTest` by changing the timeout and retries to avoid this scenario. And removed the synchronization stuff. We check later a better approach for this case.